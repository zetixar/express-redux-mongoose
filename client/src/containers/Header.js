import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { withRouter, Link } from 'react-router-dom'
import styled from 'styled-components'

import { logout } from '../actions'

const HeaderSection = styled.div`
  background-color: #181818;
  padding: 15px;
  height: 75px;
  color: #fff;
  display: flex;
  justify-content: space-between;
`
const LogoLink = styled(Link)`
  display: flex;
  align-items: center;
  font-size: 48px;
  color: #e1c348;
`

const AuthActionList = styled.ul`
  list-style-type: none;
  margin: 0px;
`

class Header extends Component {
  /*
  auth propTypes: there are three possibilites [
      null -> when app doesn't get any response back from server
      false -> when app gets falsy response back from server
      object(user info) -> when user gets authenticated successfully
    ]

  history propTypes: has to be an object with PropTypes.historyShape but I
  couldn't find the correct api for that. I need to fix that later

  */
  static propTypes = {
    logout: PropTypes.func.isRequired,
    auth: PropTypes.oneOfType([PropTypes.bool, PropTypes.shape()]),
    history: PropTypes.shape().isRequired,
  }

  static defaultProps = {
    auth: null,
  }

  renderContent() {
    switch (this.props.auth) {
      case null:
        return null
      case false:
        return (
          <li>
            <a className="btn" href="/auth/google">
              Login with Google
            </a>
          </li>
        )
      default:
        return (
          <li>
            <span style={{ marginRight: '15px' }}>
              Welcome {this.props.auth.givenName}!
            </span>
            <a
              className="btn"
              style={{ cursor: 'pointer' }}
              role="menuitem"
              tabIndex="0"
              onClick={event => {
                event.preventDefault()
                this.props.logout(this.props.history)
              }}
            >
              Logout
            </a>
          </li>
        )
    }
  }
  render() {
    return (
      <nav>
        <HeaderSection className="nav-wrapper">
          <LogoLink to={this.props.auth ? '/' : '/'}>ZeTiXaR</LogoLink>
          <AuthActionList>{this.renderContent()}</AuthActionList>
        </HeaderSection>
      </nav>
    )
  }
}

function mapStateToProps({ auth }) {
  return {
    auth,
  }
}
export default connect(mapStateToProps, { logout })(withRouter(Header))
