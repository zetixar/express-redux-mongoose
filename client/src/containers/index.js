export { default as CounterContainer } from './CounterContainer'
export { CounterContainer as CounterContainerTest } from './CounterContainer'
export { default as Header } from './Header'
export { Header as HeaderTest } from './Header'
