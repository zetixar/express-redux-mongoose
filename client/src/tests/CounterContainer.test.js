import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'

import { CounterContainer } from 'containers/CounterContainer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <CounterContainer
      increment={() => {}}
      decrement={() => {}}
      incrementIfOdd={() => {}}
      counter={0}
    />,
    div,
  )
})

it('renders consistantly[snapshot] without crashing', () => {
  const wrapper = shallow(
    <CounterContainer
      increment={() => {}}
      decrement={() => {}}
      incrementIfOdd={() => {}}
      counter={0}
    />,
  )
  expect(wrapper).toMatchSnapshot()
})
