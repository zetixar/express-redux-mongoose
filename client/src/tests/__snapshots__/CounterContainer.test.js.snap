// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders consistantly[snapshot] without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <pure(Counter)
    counter={0}
    decrement={[Function]}
    increment={[Function]}
    incrementIfOdd={[Function]}
  />,
  "nodes": Array [
    <pure(Counter)
      counter={0}
      decrement={[Function]}
      increment={[Function]}
      incrementIfOdd={[Function]}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CounterContainer
        counter={0}
        decrement={[Function]}
        increment={[Function]}
        incrementIfOdd={[Function]}
      />,
      "_debugID": 19,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CounterContainer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "decrement": [Function],
        "increment": [Function],
        "incrementIfOdd": [Function],
        "props": Object {
          "counter": 0,
          "decrement": [Function],
          "increment": [Function],
          "incrementIfOdd": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <pure(Counter)
          counter={0}
          decrement={[Function]}
          increment={[Function]}
          incrementIfOdd={[Function]}
        />,
        "_debugID": 20,
        "_renderedOutput": <pure(Counter)
          counter={0}
          decrement={[Function]}
          increment={[Function]}
          incrementIfOdd={[Function]}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CounterContainer
    counter={0}
    decrement={[Function]}
    increment={[Function]}
    incrementIfOdd={[Function]}
  />,
}
`;
